content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+In such a system, the attempts from malicious nodes to create double spending events, or in other way invalid transactions, are overwhelmed by the network votes from honest nodes.
+
+For a majority to be formed, the number of all nodes in the system must be known and its quorum calculated at any time.
+Opposed to this concept, in a Federated Byzantine Agreement System (FBAS) the participants in the network are loosely defined and can join and leave at any time without coordinating with any central authority.
+In such a system, quorums are incomplete membership snapshots that can change at any time.
+This makes the system consensus probabilistic with some error size (probability of not reaching a consensus) present at all time.
+
+The network does not trust a single entity to operate the network, but rather works as a decentralized system, in which each participant knows only about a group of others peers, but not about all the nodes in the network.
+In the FBAS, each node has knowledge of other nodes that it considers important.
+When making a decision, it waits for these important nodes to agree on a value before considering a transaction settled.
+
+Each node of that group also has knowledge about its own list of important nodes and considers their statement on a value before deciding on a transaction.
+Also each node is potentially a member of a group of nodes, that another considers a benchmarking group.
+On a higher level, the structure looks like multiple overlapping groups of participants.
+
+Eventually, when enough of the network accepts a transaction, it becomes infeasible and cannot be rolled back.
+This structure brings a high level of decentralization of decision making process among network members and mimics the behavior of social groups spreading a piece of information.
+
+Federated Byzantine Agreement is a model suitable for a worldwide consensus.
+One of the famous implementations of FBA is the [Stellar Consensus Protocol (SCP)](https://www.stellar.org/papers/stellar-consensus-protocol).
+Like non-federated Byzantine agreement, FBA addresses the problem of updating replicated state, such as transaction ledger or certificate tree and by agreeing on what updates should be applied to the state, nodes avoid contradictory values and irreconcilable states.
+
+Each state update is identified by a unique "slot", from which inter-update dependencies can be inferred.
+For example slots may be consecutively numbered positions in a sequentially applied log.
+FBA system runs a consensus protocol that ensures nodes are agreeing on slot contents.
+A node can update a value `x` in slot `i` when it has applied updates in all slots upon which `i` depends, additionally — the node believes that all correctly functioning nodes will eventually agree on value `x` for slot `i`.
+This action is called externalization — the node externalized value `x` for slot `i`.
+Outside world reacts on the externalized value — agrees or objects it.
```

Comment:
`The outside world`, `by agreeing or objecting to it`.

Reply:
This part is removed when details were removed